main.nf

```diff
@@ -347,6 +348,48 @@ process variant_effect_predictor {
 }
 
 
+process snpEff() {
+    input:
+        file vcf from vcfs_snpeff
+    output:
+        file '*snpeff_genes.txt'
+        file '*.snpeff'
+
+    publishDir params.outdir, mode: 'copy'
+
+    module 'bioinfo-tools'
+    module "$params.modules.snpeff"
+
+    // Does not use many resources, run it locally
+    executor 'local'
+
+    script:
+    """
+    vcf=$vcf ## Get rid of nextflow quoting
+    snpeffjar=''
+
+    for p in \$( tr ':' ' ' <<<"\$CLASSPATH" ); do
+        if [ -f "\$p/snpEff.jar" ]; then
+            snpeffjar="\$p/snpEff.jar"
+            break
+        fi
+    done
+    if [ -z "\$snpeffjar" ]; then
+        printf "Can't find snpEff.jar in '%s'" "\$CLASSPATH" >&2
+        exit 1
+    fi
+
+    cat "\$vcf" | sed 's/ID=AD,Number=./ID=AD,Number=R/' \
+                | vt decompose -s - \
+                | vt normalize -r $params.ref_fasta - \
+                | java -Xmx7G -jar "\$SNPEFFJAR" -formatEff -classic GRCh37.75 \
+                > "\$vcf.snpeff"
+
```

Comment:
If the pipe symbol is placed last on the line, no escaping of the newline is necessary. This may be a feature of the shells I'm using (`ksh` and `bash`).